online-banking-cz/src/test/java/com/adyen/checkout/onlinebankingcz/DefaultOnlineBankingCZDelegateTest.kt

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by atef on 14/9/2022.
+ */
+
+package com.adyen.checkout.onlinebankingcz
+
+import app.cash.turbine.test
+import com.adyen.checkout.components.model.paymentmethods.PaymentMethod
+import com.adyen.checkout.components.model.payments.request.OnlineBankingCZPaymentMethod
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.junit.jupiter.MockitoExtension
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@ExtendWith(MockitoExtension::class)
+internal class DefaultOnlineBankingCZDelegateTest {
+
+    private val delegate = DefaultOnlineBankingCZDelegate(
+        paymentMethod = PaymentMethod(),
+        paymentMethodFactory = { OnlineBankingCZPaymentMethod() }
+    )
+
+    @Nested
+    @DisplayName("when input data changes and")
+    inner class InputDataChangedTest {
+
+        @Test
+        fun `selectedIssuer is null, then output should be null`() = runTest {
+            delegate.outputDataFlow.test {
+                delegate.onInputDataChanged(OnlineBankingInputData(null))
```

Comment:
Try to follow the following structure when writing tests:
```
// GIVEN
// Perform any setup needed for the test

// WHEN
Execute the code that needs to be tested

// THEN
Do assertions
```

Pay attention here to how these 3 blocks are separated by new lines. Let's see how that would translate into this test:
```Kotlin
@Test
fun `selectedIssuer is null, then output should be null`() = runTest {
    delegate.outputDataFlow.test {
        val inputData = OnlineBankingInputData(null)
        
        delegate.onInputDataChanged(inputData)
        
        with(requireNotNull(expectMostRecentItem())) {
            assertNull(selectedIssuer)
        }
    }
}
```

Side note: `Assert.assertWhatever()` doesn't read very nicely. It is better to import the methods directly, so you just have `assertWhatever()`